simplq/src/store/asyncActions/getToken.js

```diff
@@ -0,0 +1,40 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import useAuth, { makeAuthedRequest } from 'api/auth';
+import * as RequestFactory from 'api/requestFactory';
+import { useCallback } from 'react';
+
+const typePrefix = 'getToken/action';
+let timer = null;
+const REFRESH_INTERVAL = 3000;
+
+/**
+ * A hook to access the getToken async action creator.
+ *
+ * @returns — getToken async action creator
+ */
+const useGetToken = () => {
+  const auth = useAuth();
+
+  const getToken = useCallback(
+    createAsyncThunk(typePrefix, async ({ tokenId, refresh }, { dispatch }) => {
+      if (timer) {
+        clearTimeout(timer);
+      }
+      const authedRequest = makeAuthedRequest(auth, RequestFactory.getToken(tokenId));
+      const response = await authedRequest.then((resp) => {
+        if (refresh === true) {
+          timer = setTimeout(() => dispatch(getToken({ tokenId, refresh })), REFRESH_INTERVAL);
+        }
+        return resp;
+      });
+      return response;
+    }),
+    []
+  );
+
+  return getToken;
```

Comment:
If we decide to use `useCallback()` in action, it would be better to do it in `return` statement to avoid to many nested functions.

```js
return useCallback(getToken);
```